notfound/extension.py

```diff
@@ -195,6 +195,52 @@ def process_doc(self, app, doctree):
             metadata.update({'nosearch': True})
 
 
+def handle_deprecated_configs(app, *args, **kwargs):
+    """
+    Handle deprecated configurations.
+
+    Looks for old deprecated configurations, define the new ones and triggers
+    warnings for old configs.
+    """
+    default, rebuild, types = app.config.values.get('notfound_urls_prefix')
+    if app.config.notfound_urls_prefix == default:
+        language = app.config.notfound_default_language
+        version = app.config.notfound_default_version
+        app.config.notfound_urls_prefix = '/{language}/{version}/'.format(
+            language=language,
+            version=version,
+        )
+
+    deprecated_configs = [
+        'notfound_default_language',
+        'notfound_default_version',
+        'notfound_no_urls_prefix',
+    ]
+    for config in deprecated_configs:
+        default, rebuild, types = app.config.values.get(config)
+        if getattr(app.config, config) != default:
+            message = '{config} is deprecated. Use "notfound_urls_prefix" instead.'.format(
+                config=config,
+            )
+            warnings.warn(message, DeprecationWarning, stacklevel=2)
+
+
+def validate_configs(app, *args, **kwargs):
+    """
+    Validate configs.
+
+    Shows a warning if one of the configs is not valid.
+    """
+    default, rebuild, types = app.config.values.get('notfound_urls_prefix')
+    if app.config.notfound_urls_prefix != default:
+        if not all([
+                app.config.notfound_urls_prefix.startswith('/'),
+                app.config.notfound_urls_prefix.endswith('/'),
+        ]):
+            message = 'notfound_urls_prefix should start and end with "/" (slash)'
+            warnings.warn(message, UserWarning, stacklevel=2)
```

Comment:
Again, it seems like we can just fix this for the user, instead of warning them, no?

Reply:
I don't like too much changing the value the user has defined if it's incorrect (*). I prefer to tell the user that value is incorrect instead to avoid confusions. Also, by modifying it, we need to maintain more code in our side to make something that is wrong to work "magically".

(*) considering that's a new config --the situation would be different to me, if it's about avoiding a breaking change, for example.